scripts/pylib/twister/twisterlib.py

```diff
@@ -1887,7 +1887,10 @@ class TestInstance(DisablePyTestCollectionMixin):
         out directory used is <outdir>/<platform>/<test case name>
     """
 
-    def __init__(self, testcase, platform, outdir):
+    # used for creating shorter build paths
+    shortcut_dir_counter = 0
```

Comment:
it is not our practice to add a class variable, why it can not be a instant variable?

Reply:
Yes, I agree with you, that this is not the best practice, but I need this variable, which works as a global counter (it is used to create unique folder name) and this value need to be shared between instances of `TestInstance` class. If you have better idea how to create such global variable, I am open for propositions :)

Reply:
the class instance only count in one process or twister instance, if you really care about the possible conflect, maybe use filelock would be more safe

Reply:
Yes, you are right, I do not predict this case when two Twister instances are run simultaneously (especially when `--subset` option is used). So I have to refactor this approach. Thank you for your comment!